src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs

```diff
@@ -560,29 +578,34 @@ private string capture_arguments(ChocolateyConfiguration config, PackageResult p
 
             get_environment_before(config, allowLogging: true);
 
-            foreach (var packageConfig in set_config_from_package_names_and_packages_config(config, packageInstalls).or_empty_list_if_null())
+            try
             {
-                Action<PackageResult> action = null;
-                if (packageConfig.SourceType == SourceType.normal)
+                foreach (var packageConfig in set_config_from_package_names_and_packages_config(config, packageInstalls).or_empty_list_if_null())
                 {
-                    action = (packageResult) => handle_package_result(packageResult, packageConfig, CommandNameType.install);
-                }
+                    Action<PackageResult> action = null;
+                    if (packageConfig.SourceType == SourceType.normal)
+                    {
+                        action = (packageResult) => handle_package_result(packageResult, packageConfig, CommandNameType.install);
+                    }
 
-                var results = perform_source_runner_function(packageConfig, r => r.install_run(packageConfig, action));
+                    var results = perform_source_runner_function(packageConfig, r => r.install_run(packageConfig, action));
 
-                foreach (var result in results)
-                {
-                    packageInstalls.GetOrAdd(result.Key, result.Value);
+                    foreach (var result in results)
+                    {
+                        packageInstalls.GetOrAdd(result.Key, result.Value);
+                    }
                 }
             }
-
-            var installFailures = report_action_summary(packageInstalls, "installed");
-            if (installFailures != 0 && Environment.ExitCode == 0)
+            finally
             {
-                Environment.ExitCode = 1;
-            }
+                var installFailures = report_action_summary(packageInstalls, "installed");
+                if (installFailures != 0 && Environment.ExitCode == 0)
+                {
+                    Environment.ExitCode = 1;
+                }
 
-            randomly_notify_about_pro_business(config);
+                randomly_notify_about_pro_business(config);
+            }

```

Comment:
This is a good start with install, but you'll also need this with upgrade and uninstall as well.

Reply:
This was added into the uninstall/upgrade sections as well.